components/rmrk/Collection/List/CollectionList.vue

```diff
@@ -180,12 +180,11 @@ export default class CollectionList extends mixins(
     this.fetchPageData(this.startPage)
   }
 
-  public async fetchPageData(
+  protected async fetchPageData(
```

Comment:
have you try to make it work with smartQuery? something like this
```ts
  protected fetchPageData(
    page: number,
    loadDirection = 'down'
  ): any {
    this.$apollo.addSmartQuery('collections', {
      query: collectionListWithSearch,
      loadingKey: 'isFetchingData',
      client: this.urlPrefix === 'rmrk' ? 'subsquid' : this.urlPrefix,
      manual: true,
      variables: {
        orderBy: this.searchQuery.sortBy,
        search: this.buildSearchParam(),
        listed: this.searchQuery.listed
          ? [{ price: { greaterThan: '0' } }]
          : [],
        first: this.first,
        offset: (page - 1) * this.first,
      },
      result: (res) => {
        this.handleResult(res, loadDirection)
      }
    })
  }
```

Reply:
I think it could not work with smartQuery, because it could not figure out in which `loadDirection` should be loaded when `page` change

Reply:
> SmartQuery will auto fetch data when `page` changes.

I was going to say it's the expected result since you change page url on scroll, tbh I don't really see the usefulness of having a direction

Reply:
yes I change page url on scroll, but it's unnecessary to fetch data every time. 
It only need to get data when reach the top or bottom.